### PR TITLE
Onlineddl: formalize "immediate operations", respect `--postpone-completion` strategy flag

### PR DESCRIPTION
**Pull request by shlomi-noach**
## Description

This PR formalizes what "immediate operations" are: 

Immediate operations are ones that can be performed within a split second, or rather, do not require long
running processes. Immediate operations are:
- CREATE TABLE
- DROP TABLE (which we convert into RENAME)
- All VIEW operations
- An INSTANT DDL accompanied by relevant ddl strategy flags
Non immediate operations are:
- A gh-ost migration
- A vitess (vreplication) migration

In #11898 we find that `ALTER VIEW` and instant DDLs do not respect `--postpone-completion`. This PR solves that for both, and all other, cases. Any "immediate operation" now respects that flag, including REVERTs of such operations.

`endtoend` tests added to cover cases.

## Related Issue(s)

Fixes https://github.com/vitessio/vitess/issues/11898
This PR supersedes and replaces https://github.com/vitessio/vitess/pull/11899
#6926

## Checklist

-   [ ] "Backport to:" labels have been added if this change should be back-ported
-   [x] Tests were added or are not required
-   [ ] Documentation was added or is not required

### CODE REVIEW

**Review by dbussink — APPROVED** [review]
I like how this is explicit and covers all cases instead of fixing only the ones we've immediately seen.

**Comment by dbussink** [comment]
@shlomi-noach The CI failures here seem relevant?

```
--- FAIL: TestSchemaChange/INSTANT_DDL:_postpone-completion (30.10s)
        --- FAIL: TestSchemaChange/INSTANT_DDL:_postpone-completion/expect_t1_queued (25.07s)
        --- PASS: TestSchemaChange/INSTANT_DDL:_postpone-completion/complete_t1 (5.03s)
```

Guess MySQL 5.7 is lacking support for things but the tests don't check that?

**Comment by shlomi-noach** [comment]
right! The test checks for instant DDL which is accomplishable on `8.0` but not on `5.7`. The test is correct and I'll adapt the test.

File: go/vt/vttablet/onlineddl/executor.go [hunk 1]
```diff
@@ -2178,70 +2179,137 @@ func (e *Executor) scheduleNextMigration(ctx context.Context) error {
...
 }
 
-// reviewQueuedMigrations iterates queued migrations and sees if any information needs to be updated
+// reviewEmptyTableRevertMigrations reviews a queued REVERT migration. Such a migration has the following SQL:
+// "REVERT VITESS_MIGRATION '...'"
+// There's nothing in this SQL to indicate:
+// - which table is involved?
+// - is this a table or a view?
+// - Are we reverting a CREATE? A DROP? An ALTER?
+// This function fills in the blanks and updated the database row.
```
Comment (deepthi) [inline go/vt/vttablet/onlineddl/executor.go:2188]:
```suggestion
// This function fills in the blanks and updates the database row.
```

File: go/vt/vttablet/onlineddl/executor.go [hunk 2]
```diff
@@ -2178,70 +2179,137 @@ func (e *Executor) scheduleNextMigration(ctx context.Context) error {
...
+	if err := e.updateMigrationIsView(ctx, onlineDDL.UUID, revertedRow.AsBool("is_view", false)); err != nil {
+		return false, err
+	}
+	if err := e.updateMySQLTable(ctx, onlineDDL.UUID, revertedMigration.Table); err != nil {
+		return false, err
+	}
+	return true, nil
+}
+
+// reviewImmediateOperations reviewes a queued migration and determines whether its is an "immediate operation".
```
Comment (deepthi) [inline go/vt/vttablet/onlineddl/executor.go:2230]:
```suggestion
// reviewImmediateOperations reviews a queued migration and determines whether it is an "immediate operation".
```

File: go/vt/vttablet/onlineddl/executor.go [hunk 3]
```diff
@@ -2178,70 +2179,137 @@ func (e *Executor) scheduleNextMigration(ctx context.Context) error {
...
+	}
+	if isImmediateOperation {
+		if err := e.updateMigrationSetImmediateOperation(ctx, onlineDDL.UUID); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// reviewQueuedMigrations iterates queued migrations and sees if any information needs to be updated.
```
Comment (deepthi) [inline go/vt/vttablet/onlineddl/executor.go:2266]:
```suggestion
// reviewQueuedMigrations iterates through queued migrations and sees if any information needs to be updated.
```

Comment (deepthi) [reply go/vt/vttablet/onlineddl/executor.go:2266]: or `over`

**Review by deepthi — APPROVED** [review]
I feel pretty useless as a reviewer when the only comments I have are on comments 😂 
Nice work!

**Pull request merged**